app/talk/discussion-comment.jsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import { Link } from 'react-router';
+import talkClient from 'panoptes-client/lib/talk-client';
+import Avatar from '../partials/avatar';
+import DisplayRoles from './lib/display-roles';
+import CommentBox from './comment-box';
+import commentValidations from './lib/comment-validations';
+import { getErrors } from './lib/validations';
+
+class DiscussionComment extends React.Component {
+  constructor(props) {
+    super(props);
+    this.onSubmitComment = this.onSubmitComment.bind(this);
+    this.commentValidations = this.commentValidations.bind(this);
+    this.state = {
+      commentValidationErrors: []
+    };
+  }
+
+  onSubmitComment(e, textContent, subject, reply) {
+    let comment = {
+      user_id: this.props.user.id,
+      discussion_id: +this.props.params.discussion,
+      body: textContent
+    };
+    if (!!subject && subject.id) {
+      comment = Object.assign(comment, {
+        focus_id: +subject.id,
+        focus_type: 'Subject'
+      });
+    }
+    if (reply) {
+      comment = Object.assign(comment, {
+        reply_id: reply.comment.id
+      });
+    }
+
+    return talkClient.type('comments')
+    .create(comment)
+    .save()
+    .then(this.props.onSubmitComment);
+  }
+
+  commentValidations(commentBody) {
+    // TODO: return true if any additional validations fail
+    const commentValidationErrors = getErrors(commentBody, commentValidations);
+    this.setState({ commentValidationErrors });
+    return !!commentValidationErrors.length;
+  }
+
+  render() {
+    if (!!this.props.discussion && this.props.discussion.locked) {
+      return this.lockedMessage();
```

Comment:
`this.lockedMessage` is undefined here.

This if block should probably be moved up into the parent component.